### ISSUE TITLE
Code samples are failing analysis in Travis

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/site-www#1709]
See: https://travis-ci.org/dart-lang/site-www/jobs/551280798

```
/home/travis/build/dart-lang/site-www/examples/misc/lib/effective_dart/some/other/file_2.dart is a part and cannot be analyzed.
Please pass in a library that contains this part.
WARNING: Ignoring Analyzer exit code 3
Analyzing lib, test...
  hint • This function has a return type of 'bool', but doesn't end with a return statement at lib/effective_dart/docs_good.dart:6:3 • missing_return
  hint • This function has a return type of 'String', but doesn't end with a return statement at lib/effective_dart/usage_bad.dart:45:3 • missing_return
  hint • This function has a return type of 'String', but doesn't end with a return statement at lib/effective_dart/usage_good.dart:53:3 • missing_return
3 hints found.
No analysis errors or warnings should be present in original source files.
Ensure that these issues are disabled using appropriate markers like: 
  // ignore_for_file: 2, some_analyzer_error_or_warning_id
```

**Comment by sfshaza2** [comment]
Noted. I asked @johnpryan to take a look yesterday. We are wondering if this is a result of Dart 2.4.

**Comment by kwalrath** [comment]
Fixed in #1717.
